Kernel/CMakeLists.txt

```diff
@@ -461,9 +461,15 @@ endif()
 add_executable(Kernel ${SOURCES})
 add_dependencies(Kernel generate_EscapeSequenceStateMachine.h)
 
+if(USE_CLANG_TOOLCHAIN)
+    # We need to pick up the #define __serenity__, otherwise we end up including unistd.h into the linker script
+    set(TARGET_STRING "--target=${SERENITY_CLANG_ARCH}-pc-serenity")
```

Comment:
This should be done with `CMAKE_CXX_COMPILER_TARGET`. I've got the assoicated changes in my superbuild branch :)

Reply:
How is that supposed to apply to the C(++) preprocessor? Do i write `--target=${CMAKE_CXX_COMPILER_TARGET}`?

Reply:
You have to stop thinking in terms of flags wherever you can when writing CMake code. The only places where flags are (should be) relevant in vast majority of cases are presets and toolchain files.

[CMAKE_CXX_COMPILER_TARGET](https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_COMPILER_TARGET.html) is simply a triplet that you set in the toolchain file.

Reply:
but that target still has to get somehow to our very manual preprocessor invocation

Reply:
I see that. The setup is quite messy. Maybe for now this could work, but eventually some of the options should be moved into the build system instead of relying on the compiler to preprocess files.

Personally, if this builds fine for now, then let it be and later it will be cleaned up anyway as the CMake files get cleaned up. This kind of rework would require too many changes that are outside the scope of this PR.